compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -442,6 +442,39 @@ static LogicalResult setPackConfig(func::FuncOp entryPoint,
       workgroupSizes);
 }
 
+static LogicalResult setUnPackConfig(func::FuncOp entryPoint,
+                                     tensor::UnPackOp op) {
+  // TODO(#11505): Consider multi-level tiling for handling unpack + generic
+  // cases.
+  SmallVector<int64_t> tileSizes = getDefaultWorkgroupTileSizesForPackUnPack(
+      cast<TilingInterface>(op.getOperation()),
+      /*defaultSize=*/16);
+
+  // Fixup for making tileSizes be multiple of inner_tile_sizes.
+  SmallVector<int64_t> innerTiles = op.getStaticTiles();
+  ArrayRef<int64_t> dimPos = op.getInnerDimsPos();
+  auto srcType = op.getSourceType();
+  std::array<int64_t, 3> workgroupSizes = {1, 1, 1};
+  for (auto [pos, size] : llvm::zip_equal(dimPos, innerTiles)) {
+    if (tileSizes[pos] == 0 || ShapedType::isDynamic(size)) continue;
+    tileSizes[pos] = llvm::alignTo(tileSizes[pos], size);
+    if (pos >= 3) continue;
+    // Distribute one tile on one thread.
+    if (srcType.isDynamicDim(pos)) {
+      workgroupSizes[pos] = tileSizes[pos] / size;
+    } else {
+      workgroupSizes[pos] =
+          std::min(tileSizes[pos], srcType.getDimSize(pos) * size) / size;
```

Comment:
why is the inner dim position deciding which ids do we map a loop to? To get decent performance we need to try to map the most inner dimensions to the fastest moving ids (so mapping dim N -> id.x N-1 -> id.y.
It sounds like this is not what that would do?

Also could we clearly comment on what part of this logic is needed for functionality if there is any?

Maybe I'm just not understanding this logic correctly and some extra comments are enough.

Reply:
I assumed that the rank of unpacked layout is always less than two. That's most of the cases we've seen, particularly in matmul. Let me generalize the lowering config a bit to handle other cases..